PyDSS/modes/QSTS.py

```diff
@@ -1,14 +1,25 @@
-from PyDSS.modes.solver_base import solver_base
 from datetime import timedelta
 
+from PyDSS.modes.solver_base import solver_base
+from PyDSS.utils.dss_utils import get_load_shape_resolution_secs
+
+
 class QSTS(solver_base):
     def __init__(self, dssInstance, SimulationSettings, Logger):
         super().__init__(dssInstance, SimulationSettings, Logger)
         self._dssSolution.Mode(2)
         self._dssSolution.Number(1)
         self._dssSolution.StepSize(self._sStepRes)
         self._dssSolution.MaxControlIterations(SimulationSettings['Project']['Max Control Iterations'])
-        self._dssSolution.DblHour(self._Hour + self._Second / 3600.0)
+
+        start_time_hours = self._Hour + self._Second / 3600.0
+        load_shape_resolutions_secs = get_load_shape_resolution_secs()
+        if load_shape_resolutions_secs == self._sStepRes:
+            # I don't know why this is needed in this case.
+            # The first data point gets skipped without it.
+            # FIXME
+            start_time_hours += self._sStepRes / 3600.0
+        self._dssSolution.DblHour(start_time_hours)
```

Comment:
We need to reach consensus on how to handle this.

Reply:
Let's keep what you have given it works with most common use cases. There seems to be no solution that works for all common cases.